x/incentives/keeper/distribute.go

```diff
@@ -20,22 +21,28 @@ func (k Keeper) getDistributedCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
 	return coins
 }
 
+// getToDistributeCoinsFromGauges returns coins that have not been distributed yet from the provided gauges
 func (k Keeper) getToDistributeCoinsFromGauges(gauges []types.Gauge) sdk.Coins {
 	// TODO: Consider optimizing this in the future to only require one iteration over all gauges.
+	// TODO: Lets make issue for this, requires changing getGaugesFromIterator logic in gauge.go
 	coins := k.getCoinsFromGauges(gauges)
 	distributed := k.getDistributedCoinsFromGauges(gauges)
 	return coins.Sub(distributed)
 }
```

Comment:
Make an issue to optimize getToDistributeCoinsFromGauges to only require one iteration over all gauges.

Reply:
https://github.com/osmosis-labs/osmosis/issues/2005

Reply:
should this be handled by a separate PR ?

Reply:
@catShaark I was just trying to prevent scope creep in this audit since it has already reached four separate PRs. It wasn't obvious to me how to complete this myself so I just made an issue for it so we could get the rest of these items merged. Feel free to tack issue 2005 if you have the time!